app/controllers/types_controller.rb

```diff
@@ -0,0 +1,41 @@
+class TypesController < ApplicationController
+  load_and_authorize_resource except: [:show]
+  before_action :set_type, only: [:show]
+  respond_to :js, :html
+
+  def index
+    show_all_types = Type.all.order('created_at DESC').page(params[:page]).per(4)
+    show_types_user_created = @types.order('created_at DESC').page(params[:page]).per(4)
+
+    @types = params[:show_all] == 'true' ? show_all_types : show_types_user_created
+  end
```

Comment:
@VanjaJeftic a nice way to write this would be:

```suggestion
  def index
    @types = Types.all if params[:show_all] == 'true'
    
    @types = @types.order('created_at DESC').page(params[:page]).per(4)
  end
```